src/shell.c

```diff
@@ -45,21 +46,46 @@ void exec(char *command, char *args[16], int argc) {
 int main() {
     char input[MAX_SIZE];
     char command[MAX_SIZE], *args[16];
+    char cwd[MAX_SIZE];
+
+    char *username = getlogin();
+    char hostname[MAX_SIZE];
+    gethostname(hostname, sizeof(hostname));
+
+    getcwd(cwd, sizeof(cwd));
+    printf("%s@%s:%s$ ", username, hostname, cwd);
```

Comment:
You print the path in 2 separate places, try to print it in only one place.

Reply:
Can you be more specific about the printing ? Where are the 2 separate places where this happens ?

Reply:
> Can you be more specific about the printing ? Where are the 2 separate places where this happens ?

There are actually 3 separate places where you print, try to make it into one. (Lines 56, 61, 84)